src/detect-engine.c

```diff
@@ -984,7 +986,7 @@ static void DetectBufferTypeFreeFunc(void *data)
 static int DetectBufferTypeInit(void)
 {
     BUG_ON(g_buffer_type_hash);
-    g_buffer_type_hash = HashListTableInit(256, DetectBufferTypeHashNameFunc,
+    g_buffer_type_hash = HashListTableInitWithCtx(256, DetectBufferTypeHashNameFunc,
```

Comment:
not sure I understand why in this global hash we'd free with de_ctx? That should only happen for per de_ctx hashes, right?

Reply:
This change is necessary with the Luaxform so its free function can unregister the thread ctx

Reply:
I think it shows a problem with the original code. There should be no dependency on a `de_ctx` in this global data structure. So if we need to free something using a de_ctx here, it means we're storing data in the wrong place.

Reply:
I don't think the context is required for the reasons you list. I'll rework this to not include the ctx value.